### ISSUE TITLE
False positive deprecation and availability warnings on annotated `using` alias declarations

### ISSUE THREAD

**Issue by ahatanak** [llvm/llvm-project#191099]
When a `using` alias declaration is annotated with the same attribute as its aliased type, the compiler incorrectly warns that the aliased type is unavailable/deprecated.

```
// Deprecation
struct [[deprecated]] Foo {};
using Bar [[deprecated]] = Foo;  // spurious: 'Foo' is deprecated
```

A natural use case is deprecating a chain of aliases together: if `Foo` and `Bar` were previously unannotated and you later decide to deprecate both, you should be able to annotate them without triggering warnings on the alias definitions themselves.

```
// Availability
using FeatureAlias  __attribute__((availability(macosx, introduced=11.0))) = int;
using FeatureAlias2 __attribute__((availability(macosx, introduced=11.0))) = FeatureAlias; // spurious: 'FeatureAlias' is only available on macOS 11.0 or newer
```

In both cases the annotation on the `using` declaration should suppress the diagnostic.

**Comment by shafik** [comment]
EDG has the behavior desired but gcc seems to do what we do here: https://godbolt.org/z/rWe178nsY

Seems reasonable but I have not thought it through.
